common/components/common/FeedbackModal.jsx

```diff
@@ -0,0 +1,72 @@
+// @flow
+
+import React from 'react';
+import {Modal, Button} from 'react-bootstrap';
+
+type Props = {|
+  showModal: boolean,
+  headerText: string,
+  messagePrompt: string,
+  confirmButtonText: string,
+  maxCharacterCount: number,
+  onConfirm: (boolean, string) => void,
+|};
+type State = {|
+  showModal: boolean,
+  feedbackText: string
+|};
+
+/**
+ * Generic Modal designed to solicit feedback
+ */
+class FeedbackModal extends React.PureComponent<Props, State> {
+  constructor(props: Props): void {
+    super(props);
+    this.state = {
+      showModal: false,
+      feedbackText: ""
+    }
+  }
+  
+  componentWillReceiveProps(nextProps: Props): void {
+    this.setState({ showModal: nextProps.showModal });
+  }
+  
+  onTextChange(event: SyntheticInputEvent<HTMLInputElement>): void {
+    this.state.feedbackText = event.target.value;
+    this.forceUpdate();
+  }
+  
+  confirm(confirmation: boolean): void {
+    this.props.onConfirm(confirmation, this.state.feedbackText);
+  }
+  
+  render(): React$Node {
+    return (
+      <div>
+          <Modal show={this.state.showModal}
+                 onHide={this.confirm.bind(this, false)}
+          >
+              <Modal.Header >
+                  <Modal.Title>{this.props.headerText}</Modal.Title>
+              </Modal.Header>
+              <Modal.Body>
+                {this.props.messagePrompt}
+                <div className="character-count">
+                  { (this.state.feedbackText || "").length} / {this.props.maxCharacterCount}
+                </div>
+                <textarea className="form-control" rows="4" maxLength={this.props.maxCharacterCount} value={this.state.feedbackText}
+                          onChange={this.onTextChange.bind(this)}>
+                </textarea>
+              </Modal.Body>
+              <Modal.Footer>
+                  <Button onClick={this.confirm.bind(this, false)}>Cancel</Button>
+                  <Button onClick={this.confirm.bind(this, true)}>{this.props.confirmButtonText}</Button>
```

Comment:
Might be worth using one of the [button classes](https://getbootstrap.com/docs/4.0/components/buttons/) on the confirm. `className="btn btn-success"` possibly. Same for other confirm action modals in the PR (e.g. Contact Project, Volunteer with Project) if they don't inherit from this. We should be consistent about it in modals sitewide and we're not so it may merit its own issue instead of fixing on this branch - up to you.

Reply:
Yeah, that should probably be a wider fix across the site.